src/core/agent/services/identifierService.ts

```diff
@@ -191,10 +195,30 @@ class IdentifierService extends AgentService {
     const metadata = await this.identifierStorage.getIdentifierMetadata(
       identifier
     );
+    if (metadata.groupMetadata) {
+      await this.deleteConnections(metadata.groupMetadata.groupId);
+    }
+
+    if (metadata.multisigManageAid) {
+      const localMember = await this.identifierStorage.getIdentifierMetadata(
+        metadata.multisigManageAid
+      );
+      await this.identifierStorage.updateIdentifierMetadata(
+        metadata.multisigManageAid,
+        {
+          isDeleted: true,
+        }
+      );
+      if (localMember.groupMetadata) {
```

Comment:
This should always be set. Can we just do `localMember.groupMetadata?.groupId` to avoid the undefined check?

Reply:
I'm asking to remove the unnecessary if statement.

Reply:
Can we use `localMember.groupMetadata!.groupId`? because `groupMetadata` in the interface is an optional property

Reply:
yes sure.